addons/udes_stock/models/stock_warehouse.py

```diff
@@ -6,8 +6,37 @@ class StockWarehouse(models.Model):
     _name = "stock.warehouse"
     _inherit = ["stock.warehouse", "mail.thread"]
 
+    def _domain_u_damaged_location_id(self):
+        """
+        Domain for locations outside Stock
+        """
+        Location = self.env["stock.location"]
+        stock_children_locations = Location.search(
+            [("id", "child_of", self.env.ref("stock.stock_location_stock").id)]
+        ).mapped("id")
+
+        return [("id", "not in", stock_children_locations)]
```

Comment:
Suggestion:

`return ["!", ("id", "child_of", self.env.ref("stock.stock_location_stock").id)]`

This should do the same thing, but one query instead of two.

Reply:
Happy to change but the story for this commit does not include unit tests so changing code might be a risk at this moment in time. Unless people agree it's fine.

Reply:
The test cases for the story says unit tests, so it would be within scope to add a unit test to check this if we wanted.

We should have plenty of stock and non-stock locations in our `udes_stock` tests to check.

Reply:
Sorry, you're right!

Reply:
As per discussions, unit tests are not required and your suggestion has been implemented and tested manually.